pkg/upgrade/upgrades/descriptor_utils.go

```diff
@@ -37,17 +37,18 @@ func createSystemTable(
 	return db.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
 		tKey := catalogkeys.EncodeNameKey(codec, desc)
 
+		got, err := txn.Get(ctx, tKey)
+		if err != nil {
+			return err
+		}
+		if got.Value.IsPresent() {
+			return nil
+		}
+
 		// If this descriptor doesn't have an ID, which happens for dynamic
 		// system tables, we need to allocate it an ID.
 		if desc.GetID() == descpb.InvalidID {
 			// If we're going to allocate an ID, make sure the table does not exist.
-			got, err := txn.Get(ctx, tKey)
-			if err != nil {
-				return err
-			}
-			if got.Value.IsPresent() {
-				return nil
-			}
```

Comment:
why are you getting rid of this defensive code? If you're going to get rid of it, why not get rid of the comment explaining it?

Reply:
I should delete the comment here but I just moved the code outside of the if statement, it should apply regardless of if the ID is allocated or not I believe. If the table already exists we should just exit and not re-create it.

Reply:
Move the comment too? It's weird to just leave a comment that doesn't apply to any code.